src/IO/S3/Credentials.cpp

```diff
@@ -721,4 +782,23 @@ S3CredentialsProviderChain::S3CredentialsProviderChain(
 
 }
 
+#else
+
+#    include <stdexcept>
+
+namespace DB
+{
+
+namespace S3
+{
+
+std::string getRunningAvailabilityZone()
+{
+    throw std::runtime_error("Does not support availability zone detection for non-cloud environment");
```

Comment:
throw DB::Exception with `UNSUPPORTED_METHOD`

Reply:
done. note that I lifted up the ErrorCodes::UNSUPPORTED_METHOD to the beginning otherwise style check complains about duplicate error code.